src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -133,87 +140,103 @@ private static String sanitizeParam(String param) {
 
         LOGGER.info("Run security analysis on contingency lists: {}", context.getContingencyListNames().stream().map(SecurityAnalysisWorkerService::sanitizeParam).collect(Collectors.toList()));
 
-        Mono<Network> network = getNetwork(context.getNetworkUuid(), context.getOtherNetworkUuids());
+        if (resultUuid != null && cancelComputationRequests.get(resultUuid) != null) {
+            return Mono.empty();
+        }
+
+        LOGGER.info("Loading networks");
+
+        Mono<Network> network = getNetwork(context.getNetworkUuid(), context.getOtherNetworkUuids(), resultUuid);
+
+        if (resultUuid != null && cancelComputationRequests.get(resultUuid) != null) {
```

Comment:
this "if" happens almost at the same time as the previous one, it's not needed

Reply:
Done